src/Entity/Access/RatePlanAccessControlHandler.php

```diff
@@ -0,0 +1,107 @@
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs an access control handler instance.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
+   *   The entity type definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeInterface $entity_type, EntityTypeManagerInterface $entityTypeManager) {
+    parent::__construct($entity_type);
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function createInstance(ContainerInterface $container, EntityTypeInterface $entity_type) {
+    return new static(
+      $entity_type,
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function checkAccess(EntityInterface $entity, $operation, AccountInterface $account) {
+    /** @var \Drupal\apigee_m10n\Entity\RatePlanInterface $entity */
+    $access = parent::checkAccess($entity, $operation, $account);
+
+    // Allow access if user has permission `view/purchase rate_plan as anyone`.
+    if ($account->hasPermission("$operation rate_plan as anyone")) {
+      return AccessResult::allowed();
+    }
+
+    /** @var \Apigee\Edge\Api\Monetization\Entity\RatePlanInterface $rate_plan */
+    $rate_plan = $entity->decorated();
+
+    // If rate plan is a developer category rate plan, deny access if developer
+    // does not belong to rate_plan category.
+    /** @var \Drupal\apigee_edge\Entity\DeveloperInterface $developer */
+    if ($rate_plan instanceof DeveloperCategoryRatePlanInterface) {
+      if (($category = $rate_plan->getDeveloperCategory()) && ($developer = $this->entityTypeManager->getStorage('developer')->load($account->getEmail()))) {
+        return AccessResult::allowedIf(($developer_category = $developer->decorated()->getAttributeValue('MINT_DEVELOPER_CATEGORY')) && ($category->id() === $developer_category))
+          ->andIf(AccessResult::allowedIfHasPermission($account, "$operation rate_plan"));
+      }
+      return AccessResult::forbidden("User {$developer->getEmail()} missing required developer category.");
+    }
+
+    // If rate plan is a developer rate plan, and the assigned developer is
+    // different from account, deny access.
+    if ($rate_plan instanceof DeveloperRatePlanInterface) {
+      if ($developer = $rate_plan->getDeveloper()) {
+        return AccessResult::allowedIf($account->getEmail() === $developer->getEmail())
+          ->andIf(AccessResult::allowedIfHasPermission($account, "$operation rate_plan"));
+      }
+      return AccessResult::forbidden("User {$developer->getEmail()} cannot view developer rate plan.");
+    }
+
+    return $access->andIf(AccessResult::allowedIfHasPermission($account, "$operation rate_plan"));
+  }
+
```

Comment:
This could something like: 

```
  /**
   * {@inheritdoc}
   */
  protected function checkCreateAccess(AccountInterface $account, array $context, $entity_bundle = NULL) {
    return AccessResult::forbidden('Rate plans cannot be created vie the developer portal.');
  }
```

Reply:
Thanks. Added.